mpas_analysis/shared/mpas_xarray/mpas_xarray.py

```diff
@@ -141,59 +141,114 @@ def ensure_list(alist):  # {{{
     return alist  # }}}
 
 
-def get_datetimes(ds, timestr, yearoffset):  # {{{
+def parse_time(ds, inTimeVarName, inReferenceDate, outTimeVarName,
+               outReferenceDate):  # {{{
     """
-    Computes a list of datetimes from the time variable in the dataset ds with
-    variable name (or a list of 2 names) given by timestr, typically one of
-    'daysSinceStartOfSim', 'xtime', or ['xtime_start', 'xtime_end'].
+    Converts MPAS times to `netcdftime.DatetimeNoLeap` objects consistent with
+    MPAS's 365-day calendar.
```

Comment:
Would it be possible to have 'calendar' as an input variable of parse_time, rather than hardwiring it to 365_day? This is because at the moment we are not sure what calendar will be used in future simulations. For transient simulations, for example, dates are real dates, I think, based on the gregorian calendar, so it would be nice if we could support the same types of calendar that netcdftime can support. Don't know if this complicates things even further though..